core/services/src/main/java/org/exoplatform/services/cms/documents/impl/DocumentServiceImpl.java

```diff
@@ -500,22 +505,10 @@ private String getSharedLink(Node currentNode, Node rootNode) {
       if (currentNode.isNodeType(NodetypeConstant.EXO_SYMLINK)) {
         currentNode = linkManager.getTarget(currentNode);
       }
-      List<String> path = new ArrayList<>();
-      Node targetNode = null;
-      boolean existingSymlink = false;
-      for (NodeIterator it = shared.getNodes(); it.hasNext(); ) {
-        Node node = it.nextNode();
-        path.add(((NodeImpl) node).getInternalPath().getAsString());
-        if (path.contains(((NodeImpl) shared).getInternalPath().getAsString() + "[]" + currentNode.getName() + ":1")) {
-          existingSymlink = true;
-          targetNode = node;
-          break;
-        }
-      }
-      if (existingSymlink) {
-        link = targetNode;
-      } else {
-        link = linkManager.createLink(shared, currentNode);
+      String CurrentNodeWorkspaceName = currentNode.getSession().getWorkspace().getName();
+      link = linkManager.getLinkInFolder(currentNode.getUUID(), ((NodeImpl) shared).getInternalPath().getAsString(), CurrentNodeWorkspaceName, EXO_SYMLINK);
+      if (link == null) {
+        link = linkManager.getLinkInFolder(currentNode.getUUID(), ((NodeImpl) activityNode).getInternalPath().getAsString(), CurrentNodeWorkspaceName, NT_FILE);
```

Comment:
Why ddon't you use Node.getPath() ?

Reply:
Node.getPath() returns a path in this form "groups/spaces/doecuments/.../ etc" and it doesn't match the Like query because the path is saved in this form []:1Groups[]:1spaces[]:1Documenst[]:1 etc
The right form to use and to match the like query is ((NodeImpl) activityNode).getInternalPath().getAsString() otherwise the query will not be useful

Reply:
Please recheck again
we always use this form in JCR queries **groups/spaces/doecuments/**